### PR TITLE
Add Docs for Azure OpenAI Embeddings Inference

### PR DESCRIPTION
**Pull request by markjhoy**
Add docs for the Azure OpenAI Embeddings task type.

Also updates the [Semantic Search tutorial](https://www.elastic.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html) to add in Azure OpenAI as well.

### CODE REVIEW

File: docs/reference/inference/put-inference.asciidoc [hunk 1]
```diff
@@ -266,8 +303,20 @@ maximum token length. Defaults to `END`. Valid values are:
 
 `user`:::
 (optional, string)
+<<<<<<< HEAD
```
Comment (szabosteve) [inline docs/reference/inference/put-inference.asciidoc:306]: Some conflict indicators remained in the commit. Please remove them.

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc [hunk 2]
```diff
@@ -99,4 +99,42 @@ In this example, the name of the field is `content`. It must be referenced in
...
+  }
+}
+--------------------------------------------------
+<1> The name of the field to contain the generated tokens. It must be referenced
+in the {infer} pipeline configuration in the next step.
+<2> The field to contain the tokens is a `dense_vector` field.
+<3> The output dimensions of the model. Find this value in the
+https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#embeddings-models[Azure OpenAI documentation]
+of the model you use.
+<4> The faster` dot_product` function can be used to calculate similarity
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:131]: Is it okay to keep the OpenAI reference and the docs link in this step or should it be changed to something Azure OpenAI specific?

File: docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc [hunk 3]
```diff
@@ -19,6 +19,12 @@
             id="infer-api-reindex-openai">
       OpenAI
     </button>
+    <button role="tab"
+            aria-selected="false"
+            aria-controls="infer-api-reindex-azure-openai-tab"
+            id="infer-api-reindex-azure-openai">
+      OpenAI
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc:26]:
```suggestion
      Azure OpenAI
```

File: docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc [hunk 4]
```diff
@@ -17,4 +17,12 @@ API with the HuggingFace service.
...
 
-// end::openai[]
\ No newline at end of file
+// end::openai[]
+
+// tag::azure-openai[]
+* An https://azure.microsoft.com/free/cognitive-services?azure-portal=true[Azure subscription]
+* Access granted to Azure OpenAI in the desired Azure subscription.
+You can apply for access to Azure OpenAI by completing the form at https://aka.ms/oai/access.
+* An embeddings model deployed in https://oai.azure.com/[Azure OpenAI Studio]
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc:26]:
```suggestion
* An embedding model deployed in https://oai.azure.com/[Azure OpenAI Studio].
```

File: docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc [hunk 5]
```diff
@@ -93,4 +93,33 @@ NOTE: When using this model the recommended similarity measure to use in the
...
 
-// end::openai[]
\ No newline at end of file
+// end::openai[]
+
+// tag::azure-openai[]
+
+[source,console]
+------------------------------------------------------------
+PUT _inference/text_embedding/azure-openai_embeddings <1>
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-task.asciidoc:102]: The `model_id`/`inference_id` is referenced as `azure_openai_embeddings` elsewhere.
```suggestion
PUT _inference/text_embedding/azure_openai_embeddings <1>
```

**Review by szabosteve — COMMENTED** [review]
Thanks for writing the docs! Left a couple of comments and suggestions.

File: docs/reference/inference/put-inference.asciidoc (hunk 1 above)
Comment (markjhoy) [reply docs/reference/inference/put-inference.asciidoc:306]: oof - good catch. Thanks!

File: docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc (hunk 2 above)
Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-mapping.asciidoc:131]: Are you referring to the link above (line 129)? This is the Azure-specific documentation... unless I'm missing something.

**Review by szabosteve — APPROVED** [review]
Two more tiny suggestions, otherwise docs LGTM! Thanks! 🙇

File: docs/reference/inference/put-inference.asciidoc [hunk 6]
```diff
@@ -266,8 +303,18 @@ maximum token length. Defaults to `END`. Valid values are:
...
+
+=====
++
+.`task_settings` for the `completion` task type
+[%collapsible%closed]
+=====
+`user`:::
+(optional, string)
+For `openai` service only. Specifies the user issuing the request, which can be used for abuse detection.
+>>>>>>> c5d724dcc1c (Update docs for Azure OpenAI Embeddings inference)
```
Comment (szabosteve) [inline docs/reference/inference/put-inference.asciidoc:317]:
```suggestion
```

File: docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc [hunk 7]
```diff
@@ -17,4 +17,12 @@ API with the HuggingFace service.
...
 
-// end::openai[]
\ No newline at end of file
+// end::openai[]
+
+// tag::azure-openai[]
+* An https://azure.microsoft.com/free/cognitive-services?azure-portal=true[Azure subscription]
+* Access granted to Azure OpenAI in the desired Azure subscription.
+You can apply for access to Azure OpenAI by completing the form at https://aka.ms/oai/access.
+* An embedding model deployed in https://oai.azure.com/[Azure OpenAI Studio]
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-requirements.asciidoc:26]: Nit. Just adding full stops to enforce sentence cases.

```suggestion
* An https://azure.microsoft.com/free/cognitive-services?azure-portal=true[Azure subscription].
* Access granted to Azure OpenAI in the desired Azure subscription.
You can apply for access to Azure OpenAI by completing the form at https://aka.ms/oai/access.
* An embedding model deployed in https://oai.azure.com/[Azure OpenAI Studio].
```

File: docs/reference/inference/put-inference.asciidoc (hunk 6 above)
Comment (markjhoy) [reply docs/reference/inference/put-inference.asciidoc:317]: 🤦 Thanks for the catch again... I need to check my conflict editing settings 😅

**Pull request merged**